src/metamodels/cgpm_metamodel.py

```diff
@@ -121,8 +121,9 @@ def create_generator(self, bdb, generator_id, schema_tokens, **kwargs):
             if _is_categorical(stattype):
                 qn = sqlite3_quote_name(name)
                 cursor = bdb.sql_execute('''
-                    SELECT DISTINCT %s FROM %s WHERE %s IS NOT NULL
-                ''' % (qn, qt, qn))
+                    SELECT DISTINCT %s FROM %s WHERE %s IS NOT NULL AND %s NOT
+                    IN ('', "''")
```

Comment:
I'm pretty sure that sqlite distinguishes between the (length-2) strings `''` and `""`, so perhaps this check should be something like `NOT IN ('', "''", '""')`, maybe "unbalanced" `"'` and `"'`, although I don't know whether the latter two tokens would appear between commas in any reasonable `.csv` files